linera-base/src/abi.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module defines the notion of Application Binary Interface (ABI) for Linera
+//! applications across Wasm and native architectures.
+
+use serde::{de::DeserializeOwned, Serialize};
+use std::fmt::Debug;
+
+/// A trait that includes all the types exported by a Linera application.
+pub trait ContractAbi {
+    /// Initialization argument passed to a new application on the chain that created it
+    /// (e.g. an initial amount of tokens minted).
+    ///
+    /// To share configuration data on every chain, use parameters instead.
+    type InitializationArgument: Serialize + DeserializeOwned + Send + Sync + Debug + 'static;
+
+    /// Immutable parameters specific to this application (e.g. the name of a token).
+    type Parameters: Serialize + DeserializeOwned + Send + Sync + Debug + 'static;
+
+    /// The type of operations executed by the application.
+    ///
+    /// Operations are transactions directly added to a block by the creator (and signer)
+    /// of the block. Users typically use operations to start interacting with an
+    /// application on their own chain.
+    type Operation: Serialize + DeserializeOwned + Send + Sync + Debug + 'static;
+
+    /// The type of effects executed by the application.
+    ///
+    /// Effects are executed when a message created by the same application is received
+    /// from another chain and accepted in a block.
+    type Effect: Serialize + DeserializeOwned + Send + Sync + Debug + 'static;
+
+    /// The type of arguments used to call this application from another application on the chain.
```

Comment:
nit: the argument type when this application is called from another application on the same chain.

Reply:
ok